src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -81,5 +83,31 @@ internal static ManagedIdentitySource GetManagedIdentitySource()
                 return ManagedIdentitySource.DefaultToImds;
             }
         }
+
+        // Method to return true if a file exists and is not empty to validate the Azure arc environment.
+        private static bool ValidateAzureArcEnvironment(string identityEndpoint, string imdsEndpoint)
+        {
+            if (!string.IsNullOrEmpty(identityEndpoint) || !string.IsNullOrEmpty(imdsEndpoint))
```

Comment:
Shouldn't this be && as per original logic?

Reply:
Yes, good catch. Thanks :)